##Question 2

!!!!!!!!TODO FOR YOU
what if we used the values for where the former inmate was or was not ever employed in a CEO transitional job 3 years after release (4th row of the table)?
P0=??,P1=??
look up data table and insert the appropriate values to get the appropriate result!

In [6]:
import numpy as np

In [7]:
alpha=0.05
#we like fractions better then percentages. as a rule of thumb, either use fractions or counts
P_0=0.521*0.01   ##post program 
P_1=.533*0.01

n_0=409
n_1=564

#lets get the counts by multiplying by the sample size
Nt_0=P_0*n_0
Nt_1=P_1*n_1

In [8]:
P_0=52.1 
P_1= 53.3


alpha=0.05
#we like fractions better then percentages. as a rule of thumb, either use fractions or counts
P_0=0.521*0.01 
P_1=0.701*0.01
n_0=409
n_1=564

#lets get the counts by multiplying by the sample size
Nt_0=P_0*n_0
Nt_1=P_1*n_1

#define the sample proportion first
sp=(P_0*n_0+P_1*n_1)/(n_1+n_0)
sp

# i am goonna create a little one line function to calculate the standard dev, it is not really needed, but just to show you how you do such a thing
sp_stdev= lambda p, n: np.sqrt( p * ( 1 - p ) /n[0] +  p * ( 1 - p )/n[1]  )


sp_stdev_3y=sp_stdev((Nt_0+Nt_1)/(n_0+n_1),[n_0,n_1])



zscore = lambda p0, p1, s : (p0-p1)/s
z_3y = zscore(P_1, P_0, sp_stdev_3y)


'P_0 = ',P_0 ,'n_0 = ', n_1 ,'sp_stdev_3y = ',sp_stdev_3y,'z_3y = ',z_3y




('P_0 = ',
 0.00521,
 'n_0 = ',
 564,
 'sp_stdev_3y = ',
 0.0051197664089798678,
 'z_3y = ',
 0.3515785401542677)

###z_3y =0.3515785401542677)



In [9]:
p_2y=1-0.9998


def report_result(p,a):
    print ('is the p value {0:.2f} smaller than the critical value {1:.2f}? '.format(p,a))
    if p<a:
        print('yes')
    else:  ("NO")
    
    print ('the Null hypothesis is {}'.format( 'rejected' if p<a  else 'not rejected')) 

    
report_result(p_2y,alpha)




is the p value 0.00 smaller than the critical value 0.05? 
yes
the Null hypothesis is rejected


In [10]:
Ntot = 973
expected = 82.908*48.671*481.092*360.33
sample_values = [[14.7*5.64,11.9*5.64],[85.3*4.09,88.9*4.09]]

chisqstat= lambda N, values, expect : N*((values[0][0]*values[1][1]-values[0][1]*values[1][0])**2)/(expect)

#chisqstat= lambda N, values, expect : N*((values[0][0]*values[1][1]-values[0][1]*values[1][0])**2)/(expect)

#'chisqstat = ' ,chisqstat(Ntot,  sample_values, expected)

print (chisqstat(Ntot,  sample_values, expected))



63.0048380954


###Answer : 
The final value of : 63.0048380954 is much greatere than 3.84(as alpha =0.05) which implies we reject null hypotheses because p-value>0.05.
